File: docs/snippets/all/concepts/how-does-rerun-work/log-to-grpc.py
import rerun as rr

rr.init("rerun_example_log_to_grpc")

# Connect to the Rerun gRPC server using the default address and url: rerun+http://localhost:9876/proxy
rr.connect_grpc()

# Log data as usual, thereby pushing it into the gRPC connection.
while True:
    rr.log("/", rr.TextLog("Logging things…"))

File: docs/snippets/all/concepts/how-does-rerun-work/log-to-grpc.cpp
#include <rerun.hpp>

int main(int argc, char* argv[]) {
    // Connect to the Rerun gRPC server using the default address and
    // port: localhost:9876
    const auto rec = rerun::RecordingStream("rerun_example_log_to_grpc");
    rec.connect_grpc().exit_on_failure();

    // Log data as usual, thereby pushing it into the gRPC connection.
    while (true) {
        rec.log("log", rerun::TextLog("Logging things…"));
    }
}
